### ISSUE TITLE
Top level api and spec

### ISSUE THREAD

**Issue by sumangurung** [whitequark/parser#315]
Hi, 
I couldn't find documentation on top level api and how to use this tool. Also all the specs are at the same directory level, so it was hard for me to figure out from the specs what the public api's are and how to use this tool programmatically. Is there documentation around it somewhere else? 

Also, a technical post on how the code is architected and what happens when a source is parsed would be really great.

I am happy to help if you have some ideas.

**Comment by whitequark** [comment]
Have you read [the README](https://github.com/whitequark/parser#usage) or the [API documentation](https://whitequark.github.io/parser/) which is linked from the README?

**Comment by sumangurung** [comment]
I did read both the links. I should have looked more because I found more info in the `doc` folder. This [AST Format](https://github.com/whitequark/parser/blob/master/doc/AST_FORMAT.md) in particular was helpful.

I guess, the top level api is just `parse` along with some extra config to print out diagnostics. And it can work on both a custom buffer or a plain string. For some reason, I expected it to have a bigger public api. But, this is great, I love small, simple libraries.
